src/openforms/registrations/contrib/objects_api/tests/test_backend.py

```diff
@@ -1873,3 +1871,146 @@ def match_attachment(request):
             document_create_attachment_body["vertrouwelijkheidaanduiding"], "geheim"
         )
         self.assertEqual(document_create_attachment_body["titel"], "A Custom Title")
+
+    @override_settings(ESCAPE_REGISTRATION_OUTPUT=True)
+    def test_submission_with_objects_api_escapes_html(self, m):
+        submission = SubmissionFactory.from_components(
+            [
+                {
+                    "key": "voornaam",
+                    "type": "textfield",
+                    "registration": {
+                        "attribute": RegistrationAttribute.initiator_voornamen,
+                    },
+                }
+            ],
+            submitted_data={"voornaam": "<script>alert();</script>"},
+            language_code="en",
+        )
+        submission_step = submission.steps[0]
+        step_slug = submission_step.form_step.slug
+
+        mock_service_oas_get(m, "https://objecten.nl/api/v1/", "objecten")
+        mock_service_oas_get(m, "https://documenten.nl/api/v1/", "documenten")
+
+        objects_form_options = dict(
+            objecttype="https://objecttypen.nl/api/v1/objecttypes/2",
+            objecttype_version=2,
+            productaanvraag_type="testproduct",
+            informatieobjecttype_submission_report="https://catalogi.nl/api/v1/informatieobjecttypen/2",
+            upload_submission_csv=True,
+            informatieobjecttype_submission_csv="https://catalogi.nl/api/v1/informatieobjecttypen/5",
+            organisatie_rsin="123456782",
+            zaak_vertrouwelijkheidaanduiding="geheim",
+            doc_vertrouwelijkheidaanduiding="geheim",
+        )
+
+        expected_result = {
+            "url": "https://objecten.nl/api/v1/objects/1",
+            "uuid": "095be615-a8ad-4c33-8e9c-c7612fbf6c9f",
+            "type": objects_form_options["objecttype"],
+            "record": {
+                "index": 0,
+                "typeVersion": objects_form_options["objecttype_version"],
+                "data": {
+                    "data": {
+                        f"{step_slug}": {
+                            "voornaam": "&lt;script&gt;alert();&lt;/script&gt;",
+                        }
+                    },
+                    "type": objects_form_options["productaanvraag_type"],
+                    "submission_id": str(submission.uuid),
+                    "language_code": "en",
+                    "attachments": [],
+                    "pdf_url": "https://documenten.nl/api/v1/enkelvoudiginformatieobjecten/1",
+                },
```

Comment:
this was confusing for me when reading the text, because the shape of data does not match with what is configured in the template! I'd align this with what is actually sent to the objects API.